optuna/study.py

```diff
@@ -578,6 +578,41 @@ def trials(self):
 
         return self.storage.get_all_trials(self.study_id)
 
+    @property
+    def product_search_space(self):
+        # type: () -> Dict[str, BaseDistribution]
+        """Return the product search space of the :class:`~optuna.study.InTrialStudy`.
+
+        "product search space" contains the product of parameter distributions that have been
+        suggested in the completed trials of the study so far.
+        If there are two parameters that have the same name but different distributions, neither
+        is included in the resulting search space (i.e., dynamic value space isn't supported).
```

Comment:
> "dynamic value space isn't supported" 

I think some users may think that dynamic value space cause exceptions. Can we describe the behavior directly like 'The parameters with dynamic value ranges will be excluded.' or something?

Reply:
> 'The parameters with dynamic value ranges will be excluded.'

This phrase seems good.